src/Microsoft.Extensions.Configuration.AzureAppConfiguration/OfflineFileCache.cs

```diff
@@ -237,75 +197,76 @@ private async Task DoUpdate(string tempFile, int retry = 0)
         {
-            if (azconfigOptions == null)
-            {
-                throw new ArgumentNullException(nameof(azconfigOptions));
-            }
-
-            if (azconfigOptions.ConnectionString == null)
-            {
-                throw new InvalidOperationException("An Azure App Configuration connection string is required.");
-            }
-
-            OfflineFileCacheOptions options = _options ?? new OfflineFileCacheOptions();
-
-            if ((options.Key == null) || (options.SignKey == null) || (options.IV == null))
+            if (string.IsNullOrEmpty(_appConfigEndpoint))
             {
-                byte[] secret = Convert.FromBase64String(ConnectionStringParser.Parse(azconfigOptions.ConnectionString, "Secret"));
-                using (SHA256 sha256 = SHA256.Create())
-                {
-                    byte[] hash = sha256.ComputeHash(secret);
-
-                    options.Key = options.Key ?? hash;
-                    options.SignKey = options.SignKey ?? hash;
-                    options.IV = options.IV ?? hash.Take(16).ToArray();
-                }
+                _appConfigEndpoint = (appConfigOptions.Endpoint != null)
+                                      ? appConfigOptions.Endpoint.ToString()
+                                      : ConnectionStringParser.Parse(appConfigOptions.ConnectionString, "Endpoint");
             }
-
+            
             if (string.IsNullOrEmpty(_scopeToken))
             {
-                //
                 // The default scope token is the configuration store endpoint combined with all of the key-value filters
+                var sb = new StringBuilder($"{_appConfigEndpoint}\0");
 
-                string endpoint = ConnectionStringParser.Parse(azconfigOptions.ConnectionString, "Endpoint");
-
-                if (string.IsNullOrWhiteSpace(endpoint))
-                {
-                    throw new InvalidOperationException("Invalid connection string format.");
-                }
-
-                var sb = new StringBuilder($"{endpoint}\0");
-
-                foreach (var selector in azconfigOptions.KeyValueSelectors)
+                foreach (var selector in appConfigOptions.KeyValueSelectors)
                 {
                     sb.Append($"{selector.KeyFilter}\0{selector.LabelFilter}\0");
                 }
 
                 _scopeToken = sb.ToString();
             }
 
-            _options = options;
+            if (_options.DataProtector == null)
+            {
+                IDataProtectionProvider dataProtectionProvider = DataProtectionProvider.Create($"{_options.Path}-{_appConfigEndpoint}");
+                _options.DataProtector = dataProtectionProvider.CreateProtector($"AppConfigurationOfflineFileCacheProtector-{_appConfigEndpoint}");
+            }
+
+            _timeLimitedDataProtector = _timeLimitedDataProtector ?? _options.DataProtector.ToTimeLimitedDataProtector();

```

Comment:
>_timeLimitedDataProtector = _timeLimitedDataProtector [](http://example.com/codeflow?start=12&length=53)

This kind of looks like an unintended assignment since you're assigning the same variable. If this is just for initialization I'd recommend using an if statment.